api/dash0monitoring/v1alpha1/operator_types.go

```diff
@@ -0,0 +1,173 @@
+// SPDX-FileCopyrightText: Copyright 2024 Dash0 Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	common "github.com/dash0hq/dash0-operator/api/dash0monitoring"
+)
+
+const (
+	OperatorFinalizerId = "operator.dash0monitoring.com/dash0monitoring-operator-finalizer"
+)
+
+// Dash0OperatorConfigurationSpec defines the desired state of the Dash0 monitoring resource.
+type Dash0OperatorConfigurationSpec struct {
+	// The URL of the observability backend to which telemetry data will be sent. This property is mandatory. The value
+	// needs to be the OTLP/gRPC endpoint of your Dash0 organization. The correct OTLP/gRPC endpoint can be copied fom
+	// https://app.dash0.com/settings. The correct endpoint value will always start with `ingress.` and end in
+	// `dash0monitoring.com:4317`.
+	//
+	// +kubebuilder:validation:Mandatory
+	IngressEndpoint string `json:"ingressEndpoint"`
```

Comment:
```suggestion
	endpoint string `json:"endpoint"`
```

I already renamed it in main for the Dash0Monitoring resource, we should keep it consistent.